packages/wallet-manager/src/WalletManagerController.ts

```diff
@@ -0,0 +1,162 @@
+import { ReactiveController, ReactiveControllerHost } from 'lit';
+import { EvmWallet, SubstrateWallet } from '.';
+import { Web3Provider } from '@ethersproject/providers';
+import { ApiPromise } from '@polkadot/api';
+import { AddChain } from './types';
+import { IWalletManagerController } from './interfaces';
+import { Signer } from '@ethersproject/abstract-signer';
+
+export class WalletManagerController implements IWalletManagerController {
+  host: ReactiveControllerHost;
+  evmWallet?: EvmWallet;
+  substrateWallet?: SubstrateWallet;
+  account?: string;
+  substrateAccount?: string;
+
+  constructor(host: ReactiveControllerHost) {
+    (this.host = host).addController(this as ReactiveController);
+  }
+
+  private appendProviderEvents(evmWallet: EvmWallet): void {
+    evmWallet.addListener('walletAccountChanged', (account) => {
+      this.account = account;
+      this.host.requestUpdate();
```

Comment:
reactive controllers can't use `reactive attributes` in the definition, because it would mean that they are implemeting `LitElement` which makes the constructor ilegal. Also, testing this, I realized we don't need that method, since we have public methods to attach listener and react accordingly